### PR TITLE
Rename ITS and MFT Point to Hit

### PR DESCRIPTION
**Pull request by iouribelikov**
In response to the issue https://github.com/AliceO2Group/AliceO2/issues/319 ,   here we rename the Point class to Hit everywhere inside the ITS and MFT code.   There also seem to be some commits related to the development of the MFT digitizer and MFT clusterizer.

### CODE REVIEW

File: Detectors/ITSMFT/ITS/simulation/src/Digitizer.cxx [hunk 1]
```diff
@@ -89,7 +89,7 @@ DigitContainer& Digitizer::process(TClonesArray* points)
   // Convert points to digits
   const SegmentationPixel* seg = (SegmentationPixel*)mGeometry.getSegmentationById(0);
   for (TIter pointiter = TIter(points).Begin(); pointiter != TIter::End(); ++pointiter) {
-    Point* point = dynamic_cast<Point*>(*pointiter);
+    Hit* point = dynamic_cast<Hit*>(*pointiter);
```
Comment (sawenzel) [inline Detectors/ITSMFT/ITS/simulation/src/Digitizer.cxx]: unrelated to this commit: consider using a static cast here for performance reasons.

**Comment by sawenzel** [comment]
The PR contains a lot more that Point->Hit. Is it supposed to be like this? If yes, could you update the PR message a bit?

**Comment by sawenzel** [comment]
Some commit messages appear twice. It would be nice to squash related things together.

**Comment by iouribelikov** [comment]
@sawenzel Sorry.  Indeed, it was "Point -> Hit" on the ITS side only.
@bovulpes  Bogdan, what was (in addition to "Point->Hit") done in the MFT code ?

**Comment by iouribelikov** [comment]
@sawenzel Strictly speaking, all of these commits' messages are different. (But, true, you have to read them to the "end of line" ... )

File: Detectors/ITSMFT/ITS/simulation/src/Digitizer.cxx (hunk 1 above)
Comment (iouribelikov) [inline Detectors/ITSMFT/ITS/simulation/src/Digitizer.cxx]: @sawenzel You are absolutely right.  Fixed !

**Comment by bovulpes** [comment]
In the MFT there were essential changes which would have led to conflicts if not taken into account before the Point to Hit transition. Indeed, there were two MFT commits which could have been squashed into one, but all this happened too fast.

File: Detectors/ITSMFT/MFT/reconstruction/include/MFTReconstruction/Clusterizer.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      
+    public:
+      
+      Clusterizer();
+      ~Clusterizer();
+      
+      Clusterizer(const Clusterizer&) = delete;
+      Clusterizer& operator=(const Clusterizer&) = delete;
+      
+      /// Steer conversion of points to digits
```
Comment (sawenzel) [inline Detectors/ITSMFT/MFT/reconstruction/include/MFTReconstruction/Clusterizer.h]: this comment sounds misleading. Isn't this doing digits->clusters?

File: Detectors/ITSMFT/MFT/reconstruction/include/MFTReconstruction/ClusterizerTask.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      ClusterizerTask();
+      ~ClusterizerTask() override;
+      
+      InitStatus Init() override;
+      InitStatus ReInit() override;
+      void Exec(Option_t* opt) override;
+      
+      void reset();
+      
+      virtual void initMQ(TList* tempList);
```
Comment (sawenzel) [inline Detectors/ITSMFT/MFT/reconstruction/include/MFTReconstruction/ClusterizerTask.h]: do we need this function to be virtual? Could you consider using `std::vector` for tempList? What is tempList supposed to hold?

**Comment by bovulpes** [comment]
Hello Sandro,

I agree that I have to look over the code, which I have quickly adapted from the ITS and had no time to do it before. In order to allow the ITS to continue using the new common ITSMFT part, I wonder if it it would be possible to pull these commits as they are and I will continue afterwards to look and fix the code following your remarks and searching for other weak points. The alternative is to wait, since I will not have time to do it very soon. Not the same man power in the MFT as in the ITS, sorry.

**Comment by bovulpes** [comment]
@iouribelikov 
Iouri, I could go on and fix the problems in my bovulpes/mft-Point2Hit and ask you later to update your pull request, what do you think, is the time pressing?

**Comment by bovulpes** [comment]
Hi,

I am having a look at the MFT commits right before the transition Point -> Hit. These commits were a quick copy/paste of the ITSMFT digitizer and clusterizer code and it will take some time to make it MFT-functional. I am afraid that in the meantime the code will diverge further if this pull is not applied. This is the kind of git impasse which sometimes happens with concurrent work.

**Comment by iouribelikov** [comment]
@bovulpes 
Hi Bogdan.  So what do you think would be the best ?   Should we simply close this pull request until 
the MFT digitizer/clusterizer become ready ?   And then we will open a new request ?

**Comment by bovulpes** [comment]
@iouribelikov 
Hi Iouri, yes, I think it is the best. I am not ready with the code, I will tell you when ready.

**Comment by iouribelikov** [comment]
@bovulpes  OK.  Closing this PR ...

**Pull request closed without merging**